src/mqt/templates/render_templates.py

```
@@ -263,8 +263,8 @@ def _render_template(environment: jinja2.Environment, template_container: Templa
         output = template.render(**template_container.arguments)
 
         # Write the rendered template to a file
-        output_path = output_dir / template_container.file_name
+        output_path = target_dir / template_container.output_dir / template_container.file_name
         output_path.write_text(output + "\n", encoding="utf-8")
 
         # Format the rendered template
-        subprocess.run(["prek", "run", "--files", str(output_path)], check=False)
+        subprocess.run(["prek", "run", "--files", str(output_path)], cwd=target_dir, check=False)
```

Ruff (0.15.17)

[error] `subprocess` call: check for execution of untrusted input

(S603)